### PR TITLE
added vuex store and added search filtering smartbins

### LINKED ISSUE #32: make search bar filter smartbins
the searchbar in the dashboard should work as a filter for all smartbins on the dashboard

### PR DESCRIPTION
**Pull request by willeasp**
fixes #32 
- added vuex store
- added searchquery state
- added filtering of smartbins by name and location

### CODE REVIEW

File: webapp/src/main.js [hunk 1]
```diff
@@ -18,14 +18,18 @@ import Vue from 'vue';
 import DashboardPlugin from './plugins/dashboard-plugin';
 import App from './App.vue';
 
+
```
Comment (Joppevalck) [inline webapp/src/main.js:21]: Unnecessary?

File: webapp/src/store/index.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+Vue.use(Vuex)
+
+const store = new Vuex.Store({
+  state: {
+    count: 0,
+    searchQuery: ""
+  },
+  mutations: {
+    increment (state) {
```
Comment (Joppevalck) [inline webapp/src/store/index.js:12]: Is the increment in mutation and action, getCount in getters and the count state necessary?

**Review by Joppevalck — CHANGES_REQUESTED** [review]
Increment stuff should be removed

File: webapp/src/store/index.js (hunk 2 above)
Comment (willeasp) [reply webapp/src/store/index.js]: fixed!

**Review by Joppevalck — APPROVED** [review]

**Pull request merged**